crates/extensions/c8y_mapper_ext/src/tests.rs

```diff
@@ -1425,6 +1425,8 @@ async fn mapper_dynamically_updates_supported_operations_for_tedge_device() {
 }
 
 #[tokio::test]
+// TODO: fix or remove test
+#[ignore = "asserts that publishing a single operation capability message causes full rescan of c8y operations directory, which is undesirable behaviour"]
```

Comment:
I understand that the rescan is not optimal. But, we shouldn't be skipping the whole test for that reason. The dynamic ops update feature is still relevant and must be tested.

Reply:
Yeah, this will need to be addressed before the merge.
Just that from my perspective, it looks like the primary point of this test is to assert the exact behaviour I'm removing (registering any operation via MQTT causes full rescan of c8y operations directory), so I'm not really sure what behaviour we could be checking instead.

Reply:
Okay, I discovered the reason why the test checked that behaviour (#2614), and found that my implementation was incorrect, and in fact, we need to still maintain the old behaviour for child devices.

Fixed in a313601